test/utils/testUtils.js

```diff
@@ -75,12 +75,24 @@ utils.deleteTable = function (db, tableName) {
 }
 
 /*
+ * A helper to generate an index name for testing indices.
+ *
+ * @param hashKey {string}
+ * @param rangeKey {string}
+ */
+utils.indexNameGenerator = function (hashKey, rangeKey) {
+  var name = 'index-' + hashKey
+  if (rangeKey) name = name + '-' + rangeKey
+  return name
+}
+
+/*
  * A helper function that creates the testing table.
  *
  * @param db {AWS.DynamoDB} The database instance.
  * @return {Promise}
  */
-utils.createTable = function (db, tableName, hashKey, rangeKey) {
+utils.createTable = function (db, tableName, hashKey, rangeKey, gsiDefinitions) {
```

Comment:
i will, but i'll do it as a follow up if that's ok. we need the GSI support for a product feature in the short term.